src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts

```diff
@@ -41,22 +41,20 @@ export default function useClientSideSmartOrderRouterTrade<TTradeType extends Tr
   state: TradeState
   trade: InterfaceTrade<Currency, Currency, TTradeType> | undefined
 } {
+  const amount = useMemo(() => amountSpecified?.asFraction, [amountSpecified])
+  const [currencyIn, currencyOut] =
```

Comment:
are we not memoizing this on purpose?

Reply:
I don't think we gain anything by memoizing this, because it's just a boolean check.